src/decorators/attachment/AttachmentExtension.ts

```diff
@@ -0,0 +1,33 @@
+import { Expose, Type } from 'class-transformer'
+import { IsArray, IsOptional, ValidateNested } from 'class-validator'
+
+import { BaseMessageConstructor } from '../../agent/BaseMessage'
+import { Attachment } from './Attachment'
+
+export function AttachmentDecorated<T extends BaseMessageConstructor>(Base: T) {
+  class AttachmentDecoratorExtension extends Base {
+    /**
+     * The ~attach decorator is required for appending attachments to a credential
+     */
+    @Expose({ name: '~attach' })
+    @Type(() => Attachment)
+    @ValidateNested()
+    @IsArray()
+    @IsOptional()
+    public attachments?: Attachment[]
+
+    public getAttachmentById(id: string): Attachment | undefined {
+      return this.attachments?.find((attachment) => attachment.id === id.substring(0, 64))
```

Comment:
We shouldn't truncate the `id` property here, it should just match the two string values.

Truncating can be done on a higher level